src/constructor_helpers.cpp

```diff
@@ -474,3 +497,33 @@ bool model_reservoir(Reservoir *reservoir, Reservoir_KML_Coordinates *coordinate
 
   return true;
 }
+
+bool model_bulk_pit(Reservoir *reservoir, Reservoir_KML_Coordinates *coordinates,
+                     vector<vector<vector<GeographicCoordinate>>> &countries,
+                     vector<string> &country_names, std::vector<PitCharacteristics> pit_shapes) {
+  
+  vector<GeographicCoordinate> pit_polygon;
+  
+  for (uint i = 0; i < pit_shapes.size(); i++) {
+    if (reservoir->identifier == pit_shapes[i].res_identifier)
+      pit_polygon = pit_shapes[i].brownfield_polygon;
+  }
+
+  string polygon_string = str(compress_poly(corner_cut_poly(pit_polygon)), reservoir->elevation);
+  coordinates->reservoir = polygon_string;
+  
+  GeographicCoordinate origin = get_origin(reservoir->latitude, reservoir->longitude, border);
+  reservoir->shape_bound.clear();
+  for(GeographicCoordinate p : pit_polygon)
+    reservoir->shape_bound.push_back(convert_coordinates(p, origin));
+  
+  //KML
+  for(uint i = 0; i< countries.size();i++){
+      if(check_within(GeographicCoordinate_init(reservoir->latitude, reservoir->longitude), countries[i])){
```

Comment:
Simplified this function since shapes are now stored in the pair object. Still need this to be separated from model_existing_reservoirs since there is no Shapefile input for bulk_pits after screening (Shapefiles are only used for the whole mining tenaments, not the pits themselves).